xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp

```diff
@@ -330,14 +355,6 @@ void CPeripheralBusAddon::GetDirectory(const std::string &strPath, CFileItemList
     addon->GetDirectory(strPath, items);
 }
 
-void CPeripheralBusAddon::OnEvent(const ADDON::AddonEvent& event)
-{
-  if (typeid(event) == typeid(ADDON::AddonEvents::Enabled) ||
-      typeid(event) == typeid(ADDON::AddonEvents::Disabled) ||
-      typeid(event) == typeid(ADDON::AddonEvents::InstalledChanged))
-    UpdateAddons();
```

Comment:
> This with "ADDON::AddonEvents::InstalledChanged" has not worked for updated (only install and deinstall).
>
> A little bit of the problem of himself, but he compares installed addon id's with those already known to him.

I know, but this is an issue with CPeripheralBusAddon::UpdateAddons only (and it's still broken) as it just looks at added/removed addon ids and does not account for updates. The fix is to remove this stuff from UpdateAddons and instead first unload the addon with that id completely, then try to load addon again, when an addon reinstall occur.